Sources/Public/Item Views/DayView.swift

```diff
@@ -0,0 +1,344 @@
+      addInteraction(UIPointerInteraction(delegate: self))
+    }
+  }
+
+  @available(*, unavailable)
+  required init?(coder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+
+  // MARK: Public
+
+  public override func layoutSubviews() {
+    super.layoutSubviews()
+
+    let edgeInsets = invariantViewProperties.edgeInsets
+    let insetBounds = bounds.inset(
+      by: UIEdgeInsets(
+        top: edgeInsets.top,
+        left: edgeInsets.leading,
+        bottom: edgeInsets.bottom,
+        right: edgeInsets.trailing))
+
+    let path: CGPath
+    switch invariantViewProperties.shape {
+    case .circle:
+      path = UIBezierPath(
+        ovalIn: CGRect(
+          origin: CGPoint(x: edgeInsets.leading, y: edgeInsets.top),
+          size: insetBounds.size)).cgPath
+
+    case .roundRect(let cornerRadius):
+      path = UIBezierPath(roundedRect: insetBounds, cornerRadius: cornerRadius).cgPath
+    }
+
+    backgroundLayer.path = path
+    highlightLayer?.path = path
+
+    label.frame = CGRect(
+      x: edgeInsets.leading,
+      y: edgeInsets.top,
+      width: insetBounds.width,
+      height: insetBounds.height)
+  }
+
+  public override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
+    super.touchesBegan(touches, with: event)
+
+    isHighlighted = true
+
+    if
+      case let .enabled(playsHapticsOnTouchDown, _) = invariantViewProperties.interaction,
+      playsHapticsOnTouchDown
+    {
+      feedbackGenerator = UISelectionFeedbackGenerator()
+      feedbackGenerator?.prepare()
+    }
+  }
+
+  public override func touchesMoved(_ touches: Set<UITouch>, with event: UIEvent?) {
+    super.touchesMoved(touches, with: event)
+
+    if let touchLocation = touches.first?.location(in: self), bounds.contains(touchLocation) {
+      isHighlighted = true
+    } else {
+      isHighlighted = false
+    }
+  }
+
+  public override func touchesEnded(_ touches: Set<UITouch>, with event: UIEvent?) {
+    super.touchesEnded(touches, with: event)
+
+    isHighlighted = false
+
+    feedbackGenerator?.selectionChanged()
+    feedbackGenerator = nil
+  }
+
+  public override func touchesCancelled(_ touches: Set<UITouch>, with event: UIEvent?) {
+    super.touchesCancelled(touches, with: event)
+
+    isHighlighted = false
+    feedbackGenerator = nil
+  }
+
+  // MARK: Fileprivate
+
+  fileprivate func setViewModel(_ viewModel: ViewModel) {
+    label.text = viewModel.dayText
+
+    accessibilityLabel = viewModel.accessibilityLabel
+    accessibilityHint = viewModel.accessibilityHint
+  }
+
+  // MARK: Private
+
+  private let invariantViewProperties: InvariantViewProperties
+  private let backgroundLayer: CAShapeLayer
+  private let highlightLayer: CAShapeLayer?
+  private let label: UILabel
+
+  private var feedbackGenerator: UISelectionFeedbackGenerator?
+
+  private var isHighlighted: Bool {
+    get { !(highlightLayer?.isHidden ?? true) }
+    set { highlightLayer?.isHidden = !newValue }
```

Comment:
In the future some animation here would be nice

Reply:
Added!

Reply:
Commit for it is https://github.com/airbnb/HorizonCalendar/pull/140/commits/5d079312654a2e9aa0d4d4a1ea9036b8d97c36dd